mirrord/operator/src/client.rs

```diff
@@ -649,14 +624,44 @@ impl OperatorApi<PreparedClientCert> {
     {
         self.check_feature_support(config)?;
 
+        // Check if the deplyment is empty if so then
+        let is_empty_deployment = match &config.target.path {
+            Some(Target::Deployment(target)) => {
+                let deployment_api = get_k8s_resource_api::<Deployment>(
+                    &self.client,
+                    config.target.namespace.as_deref(),
+                );
+
+                let deployment = deployment_api
+                    .get(&target.deployment)
+                    .await
+                    .map_err(|error| OperatorApiError::KubeError {
+                        error,
+                        operation: OperatorOperation::FindingTarget,
+                    })?;
+
+                !deployment
+                    .status
+                    .map(|status| status.available_replicas > Some(0))
+                    .unwrap_or_default()
+            }
+            _ => false,
+        };
+
         let target = if config.feature.copy_target.enabled
             // use copy_target for splitting queues
             || config.feature.split_queues.is_set()
+            || is_empty_deployment
         {
+            let mut copy_subtask = progress.subtask("copying target");
+
             if config.feature.copy_target.enabled.not() {
-                tracing::info!("Creating a copy-target for queue-splitting (even thought copy_target was not explicitly set).")
+                if is_empty_deployment.not() {
```

Comment:
Welcome to the `.not()` side dima!

Reply:
![](https://media4.giphy.com/media/v1.Y2lkPTc5MGI3NjExeTB0eXNlMDFodDJtamx5MmY4M2h0MmRva3p1azZ6d3BxOTF6YjNyZyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/10ORk7l5EDa4JG/200.gif)